### ISSUE TITLE
Error connecting to text-generation-webui and AllTalk v2 standalone

### ISSUE THREAD

**Issue by ShaunCassidyPoster** [semperai/amica#138]
Hi! I've just installed Amica and am trying to use it locally with Oobabooga's webui and Alltalk v2 standalone, both newly updated/installed, on a Windows 10 system. Sending a message to the webui API yields an error in the Amica interface and in the webui console.

I am able to connect SillyTavern to these very same running instances without error, so that suggests Amica is sending something incorrectly or I've misconfigured something.

**Amica error:**

```
1 of 1 unhandled error

Unhandled Runtime Error
TypeError: Cannot read properties of undefined (reading 'split')

Call Stack
basename
webpack-internal:/src/components/settings/common.tsx (128:20)
handleVRMAnimationEvent
webpack-internal:/src/features/amicaLife/eventHandler.ts (51:82)
handleIdleEvent
webpack-internal:/src/features/amicaLife/eventHandler.ts (180:19)
AmicaLife.processingEvent
webpack-internal:/src/features/amicaLife/amicaLife.ts (150:108)
```

**Webui error:**

```
23:57:23-231506 ERROR    Could not find the character "None" inside characters/. No character has been loaded.
Exception in ASGI application
Traceback (most recent call last):
  File "F:\text-generation-webui\installer_files\env\Lib\site-packages\sse_starlette\sse.py", line 247, in __call__
    await wrap(partial(self.listen_for_disconnect, receive))
  File "F:\text-generation-webui\installer_files\env\Lib\site-packages\sse_starlette\sse.py", line 236, in wrap
    await func()
  File "F:\text-generation-webui\installer_files\env\Lib\site-packages\sse_starlette\sse.py", line 191, in listen_for_disconnect
    message = await receive()
              ^^^^^^^^^^^^^^^
  File "F:\text-generation-webui\installer_files\env\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 555, in receive
    await self.message_event.wait()
  File "F:\text-generation-webui\installer_files\env\Lib\asyncio\locks.py", line 213, in wait
    await fut
asyncio.exceptions.CancelledError: Cancelled by cancel scope 25c79203890

During handling of the above exception, another exception occurred:

  + Exception Group Traceback (most recent call last):
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 401, in run_asgi
  |     result = await app(  # type: ignore[func-returns-value]
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 60, in __call__
  |     return await self.app(scope, receive, send)
  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\fastapi\applications.py", line 1054, in __call__
  |     await super().__call__(scope, receive, send)
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\starlette\applications.py", line 113, in __call__
  |     await self.middleware_stack(scope, receive, send)
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\starlette\middleware\errors.py", line 187, in __call__
  |     raise exc
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\starlette\middleware\errors.py", line 165, in __call__
  |     await self.app(scope, receive, _send)
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\starlette\middleware\cors.py", line 93, in __call__
  |     await self.simple_response(scope, receive, send, request_headers=headers)
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\starlette\middleware\cors.py", line 144, in simple_response
  |     await self.app(scope, receive, send)
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\starlette\middleware\exceptions.py", line 62, in __call__
  |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\starlette\_exception_handler.py", line 62, in wrapped_app
  |     raise exc
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\starlette\_exception_handler.py", line 51, in wrapped_app
  |     await app(scope, receive, sender)
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\starlette\routing.py", line 715, in __call__
  |     await self.middleware_stack(scope, receive, send)
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\starlette\routing.py", line 735, in app
  |     await route.handle(scope, receive, send)
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\starlette\routing.py", line 288, in handle
  |     await self.app(scope, receive, send)
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\starlette\routing.py", line 76, in app
  |     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\starlette\_exception_handler.py", line 62, in wrapped_app
  |     raise exc
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\starlette\_exception_handler.py", line 51, in wrapped_app
  |     await app(scope, receive, sender)
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\starlette\routing.py", line 74, in app
  |     await response(scope, receive, send)
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\sse_starlette\sse.py", line 233, in __call__
  |     async with anyio.create_task_group() as task_group:
  |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\anyio\_backends\_asyncio.py", line 763, in __aexit__
  |     raise BaseExceptionGroup(
  | ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\sse_starlette\sse.py", line 236, in wrap
    |     await func()
    |   File "F:\text-generation-webui\installer_files\env\Lib\site-packages\sse_starlette\sse.py", line 221, in stream_response
    |     async for data in self.body_iterator:
    |   File "F:\text-generation-webui\extensions\openai\script.py", line 129, in generator
    |     for resp in response:
    |   File "F:\text-generation-webui\extensions\openai\completions.py", line 549, in stream_chat_completions
    |     for resp in chat_completions_common(body, is_legacy, stream=True):
    |   File "F:\text-generation-webui\extensions\openai\completions.py", line 334, in chat_completions_common
    |     for a in generator:
    |   File "F:\text-generation-webui\modules\chat.py", line 403, in generate_chat_reply
    |     for history in chatbot_wrapper(text, state, regenerate=regenerate, _continue=_continue, loading_message=loading_message, for_ui=for_ui):
    |   File "F:\text-generation-webui\modules\chat.py", line 346, in chatbot_wrapper
    |     prompt = apply_extensions('custom_generate_chat_prompt', text, state, **kwargs)
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "F:\text-generation-webui\modules\extensions.py", line 231, in apply_extensions
    |     return EXTENSION_MAP[typ](*args, **kwargs)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "F:\text-generation-webui\modules\extensions.py", line 107, in _apply_custom_generate_chat_prompt
    |     return extension.custom_generate_chat_prompt(text, state, **kwargs)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "F:\text-generation-webui\extensions\superboogav2\script.py", line 169, in custom_generate_chat_prompt
    |     return custom_generate_chat_prompt_internal(user_input, state, collector, **kwargs)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "F:\text-generation-webui\extensions\superboogav2\chat_handler.py", line 105, in custom_generate_chat_prompt_internal
    |     history_as_text = _concatinate_history(kwargs['history'], state)
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "F:\text-generation-webui\extensions\superboogav2\chat_handler.py", line 58, in _concatinate_history
    |     user_name, bot_name = _get_names(state)
    |                           ^^^^^^^^^^^^^^^^^
    |   File "F:\text-generation-webui\extensions\superboogav2\chat_handler.py", line 51, in _get_names
    |     user_name, bot_name, _, _, _ = load_character_memoized(character, user_name, '')
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "F:\text-generation-webui\modules\chat.py", line 793, in load_character_memoized
    |     return load_character(character, name1, name2)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "F:\text-generation-webui\modules\chat.py", line 739, in load_character
    |     raise ValueError
    | ValueError
    +------------------------------------
```

To replicate the error (and assure myself I hadn't misconfigured things), I started over from the beginning and set up Amica anew, with the default prompt and Amica-A model. Trying to send a message produced the same errors in the webui and in Amica...

**Amica error on second try:**
```
Unhandled Runtime Error
TypeError: Cannot read properties of undefined (reading 'split')

Source
src\components\settings\common.tsx (88:17) @ split

  86 | 
  87 | export function basename(path: string) {
> 88 | const a = path.split("/");
     |               ^
  89 | return a[a.length - 1];
  90 | }
  91 | 
Call Stack
randomAnimation
src\features\amicaLife\eventHandler.ts (59:20)
handleVRMAnimationEvent
src\features\amicaLife\eventHandler.ts (256:12)
idleEvent
src\features\amicaLife\amicaLife.ts (218:32)
```

The upshot is that I haven't yet been able to get Amica to reply, but otherwise everything seems to check out. I'd appreciate any advice. Thanks!

**Comment by flukexp** [comment]
have you tried this solution ? [None Character Error #4320](https://github.com/oobabooga/text-generation-webui/issues/4320#issuecomment-2146715010).

**Comment by ShaunCassidyPoster** [comment]
Thanks! I made the webui edit first and tried again. Now I don't see that error--or any webui console activity at all--when sending Amica a message. I do get the same "split" error.

My src/paths.ts is indeed blank:
```
export const bgImages = [];
export const vrmList = [];
export const speechT5SpeakerEmbeddingsList = [];
export const animationList = [];
```
I have been selecting the VRM manually. The character appears, blinking and swaying slightly. List boxes in the "Appearance" settings are not populated, things like "Character Animations" or "Background Image". A default background image appears and I can select them manually, but I suspect it's supposed to show thumbnails of the backgrounds included in public/bg/.

Do you know how I might generate the paths in src/paths.ts as shown in your image?

**Comment by flukexp** [comment]
Run this command. <br> <code>npm run generate:paths

**Comment by ShaunCassidyPoster** [comment]
Hi flukexp, When I try the command alone, it seems to run, as below. It changes the modified date of paths.ts, but the content of the file remains the same, with the empty brackets. The file isn't read-only. That sequence also seems to run during the ```npm run dev``` startup, too. I see it going by on the console and paths.ts gets its timestamp modified, but no paths are added.

```
PS F:\amica> npm run generate:paths

> amica@0.1.0 generate:paths
> npx node scripts/generate_paths.js

Need to install the following packages:
node@23.1.0
Ok to proceed? (y) y

PS F:\amica>
```

**Comment by flukexp** [comment]
Ensure the following folders in the `public/` directory contain the respective files. The folders are:

1. `animations`
   - **Example files**:
     - `dance.vrma`
     - `greeting.vrma`
     - `idle_loop.vrma`
     - `modelPose.vrma`
     - `peaceSign.vrma`
     - `shoot.vrma`
     - `showFullBody.vrma`
     - `spin.vrma`
     - `squat.vrma`

2. `bg` (Background Images)
   - **Example files**:
     - `bg-forest1`
     - Other background images as required

3. `speecht5_speaker_embeddings`
   - **Example files**:
     - `cmu_us_awb-arctic-wab-arctic_a2002.bin`
     - `cmu_us_slt_arctic-wav-arctic_a0001.bin`

4. `vrm`
   - **Example files**:
     - `AvatarSample_A.vrm`
     - `AvatarSample_B.vrm`
     - Other VRM files as required

### Alternative: Manually Add Paths in `path.ts`

If files are missing, you can add paths manually in your `path.ts` as follows:

```typescript
export const bgImages = [
  "/bg/bg-arbius1.jpg",
  "/bg/bg-arbius2.jpg",
  "/bg/bg-forest1.jpg",
  "/bg/bg-landscape1.jpg",
  "/bg/bg-landscape2.jpg",
  "/bg/bg-landscape3.jpg",
  "/bg/bg-room1.jpg",
  "/bg/bg-room2.jpg",
  "/bg/bg-sunset1.jpg",
  "/bg/bg-town1.jpg"
];

export const vrmList = [
  "/vrm/AvatarSample_A.vrm",
  "/vrm/AvatarSample_B.vrm",
  "/vrm/AvatarSample_C.vrm",
  "/vrm/AvatarSample_D.vrm"
];

export const speechT5SpeakerEmbeddingsList = [
  "/speecht5_speaker_embeddings/cmu_us_awb_arctic-wav-arctic_a0002.bin",
  "/speecht5_speaker_embeddings/cmu_us_slt_arctic-wav-arctic_a0001.bin"
];

export const animationList = [
  "/animations/dance.vrma",
  "/animations/greeting.vrma",
  "/animations/idle_loop.vrma",
  "/animations/modelPose.vrma",
  "/animations/peaceSign.vrma",
  "/animations/shoot.vrma",
  "/animations/showFullBody.vrma",
  "/animations/spin.vrma",
  "/animations/squat.vrma"
];
```

**Comment by ShaunCassidyPoster** [comment]
I manually edited paths.ts as shown above and started Amica. Shortly into the startup process, all of the values get removed. Could the path generation script be misbehaving? (I'm not an experienced programmer, but a putterer.) Notepad++ alerts about the change when its window is next focused. Everything between brackets is dropped, as below. Incidentally, I installed by cloning the repository as described on the readme for local use. Thanks!

```
export const bgImages = [];
export const vrmList = [];
export const speechT5SpeakerEmbeddingsList = [];
export const animationList = [];
```

**Comment by flukexp** [comment]
I believe this might be because the public folder’s subfolders—animations, bg, speecht5_speaker_embeddings, and vrm—do not contain the necessary files. Could you check these folders as I described ?

<img width="252" alt="Screenshot 2567-10-26 at 21 23 51" src="https://github.com/user-attachments/assets/6b8c0dca-fab7-46ba-bb19-2180dd46723b">

**Comment by ShaunCassidyPoster** [comment]
Hi! Yes, all the files are present and in good working order. There are jpegs and thumbnails, the vrm files can be imported to Blender, etc. I tallied each file according to the list to make sure. I just ran Amica again and when "generate:paths" appeared in the console (one of the first things that happens, I'm guessing), I clicked over to Notepad++ to find that the entries had already been erased (an alert dialog mentions a change and provides a button to reload the file).

**Comment by flukexp** [comment]
There are several possible cases where this code might generate empty lists:

- **Inconsistent Path Separator**: On Windows, using forward slashes (`/`) in paths might lead to issues in some environments.

- **File System Permissions**: If the script lacks read permission for the directories it’s attempting to access, `globSync` will silently return empty lists. Verifying permissions can help ensure the paths are generated correctly.

As a workaround, you can manually add paths to `paths.ts`, as you’ve done before, and then modify the `dev` script in the `package.json` file to only run `next dev`. I have tried this solution in my environment, and it works well: removing `npm run generate:paths` from the script avoids recompiling `paths.ts` each time the server starts, allowing you to keep your manually edited `paths.ts` file. Hopefully, this helps solve the issue! 😊

```json
{
  "name": "amica",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev"
  }
}
```

**Comment by ShaunCassidyPoster** [comment]
Hi! I did get Amica to work with your advice. First, I had tried setting windowsPathsNoEscape for each of the entries in generate_paths.js, but I didn't get it to work (and I don't think fast-glob lets it be set globally).

I encountered a significant number of font retrieval errors that were fixed by changing the dev timeout in /amica/node_modules/next/dist/compiled/@next/font/dist/google/fetch-font-file.js. My internet might suck.

```
//  const timeoutId = setTimeout(() => controller.abort(), 3000);
    const timeoutId = setTimeout(() => controller.abort(), 30000);
```

I had a little difficulty hitting AllTalk 2 standalone properly. It did work, but without the option to select an XTTSv2 "priming" sample wav, I could only get a default voice (though it used the cadence of my fine-tuned checkpoint). The words came eventually, but I didn't see the VRM model's mouth moving. (To be fair, my 3060 12GB was about topped off.)

It will take some time to learn how the autonomous features work. I clicked the import button on the left side and it started downloading some *.cabs or something, while I expected to be able to import stuff to be seen when clicking on the brain icon.

Thanks a lot for your help! Amica's pretty cool.
